dace/autodiff/library/torch_integration.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
+"""
+Hooks for PyTorch tensors to make them compatible with dace
+"""
+
+try:
+    import torch
+    TORCH_AVAILABLE = True
+except ImportError:
+    torch = None
+    TORCH_AVAILABLE = False
+
+from dace import data
+
+from dace.autodiff.library.library import ParameterArray
+
+if TORCH_AVAILABLE:
+
+    def create_descriptor_tensor(self: torch.Tensor) -> data.Data:
+        """
+        Creates a descriptor for a tensor.
+        If the tensor requires grad, we convert to a ParameterArray
+        """
+
+        desc = data.create_datadescriptor(self, no_custom_desc=True)
+        if not isinstance(desc, data.Array):
+            raise ValueError("Unsupported descriptor: {}".format(desc))
+
+        if not self.requires_grad:
+            return desc
+        # initialize with no gradient buffer
+        new_desc = ParameterArray(desc.dtype,
```

Comment:
you might want to use `return ParameterArray.make_parameter(desc)` or however you called the static method.